types/mempool/priority_nonce.go

```diff
@@ -413,13 +444,13 @@ func (mp *PriorityNonceMempool) Remove(tx sdk.Tx) error {
 	return nil
 }
 
-func IsEmpty(mempool Mempool) error {
-	mp := mempool.(*PriorityNonceMempool)
+func IsEmpty[C comparable](mempool Mempool) error {
+	mp := mempool.(*PriorityNonceMempool[C])
 	if mp.priorityIndex.Len() != 0 {
 		return fmt.Errorf("priorityIndex not empty")
 	}
 
-	var countKeys []int64
+	var countKeys []C
```

Comment:
I'm not sure when `IsEmpty` is called outside of tests, but since we know the resulting size of `countKeys`, it can be pre-sized appropriately to reduce garbage creation: `countKeys := make([]C, 0, len(mp.priorityCounts))`.